app/controllers/AuthorizedToViewController.scala

```diff
@@ -69,56 +69,180 @@ class AuthorizedToViewController @Inject()(authenticate: IdentifierAction,
-            Future.successful(BadRequest(authorisedToViewSearch(form.withError("value",
-              "cf.account.authorized-to-view.search-own-accountnumber").fill(query),
-                Some(""), LocalDate.now.toString, false)))
+      query => processSearchQuery(request, query)
+    )
+  }
+
+  private def processSearchQuery(request: AuthenticatedRequest[AnyContent],
+                                 query: EORI)(
+                                  implicit hc: HeaderCarrier,
+                                  messages: Messages, appConfig: AppConfig): Future[Result] = {
+    val searchQuery = stripWithWhitespace(query)
+
+    val result = for {
+      cdsAccounts: CDSAccounts <- apiService.getAccounts(request.user.eori)
+      xiEORI: Option[EORI] <- dataStoreService.getXiEori(request.user.eori)
+    } yield {
+      val isMyAcc = cdsAccounts.myAccounts.exists(_.number == query)
+
+      (request.user.eori, isMyAcc) match {
+        case (eori, _) if eori.equalsIgnoreCase(query) =>
+          Future.successful(
+            BadRequest(authorisedToViewSearch(
+              form.withError("value", "cf.account.authorized-to-view.search-own-eori").fill(query),
+              Some(""),
+              LocalDate.now.toString,
+              fileExists = false)(request, messages, appConfig)))
+        case (_, true) =>
+          Future.successful(
+            BadRequest(authorisedToViewSearch(
+              form.withError("value", "cf.account.authorized-to-view.search-own-accountnumber").fill(query),
+              Some(""),
+              LocalDate.now.toString,
+              fileExists = false)(request, messages, appConfig)))
+        case _ =>
+          if(isSearchQueryAnAccountNumber(searchQuery)) {
+            searchAuthoritiesForValidInput(request, searchQuery, xiEORI)
+          } else {
+            searchAuthoritiesForValidInput(request, searchQuery)
           }
-        else {
-          apiService.searchAuthorities(request.user.eori, searchQuery).flatMap {
-            case Left(NoAuthorities) => Future.successful(Ok(authorisedToViewSearchNoResult(searchQuery)))
-            case Left(SearchError) => Future.successful(InternalServerError(errorHandler.technicalDifficulties))
-            case Right(searchedAuthorities) => {
-
-              val displayLink: Boolean = getDisplayLink(searchedAuthorities)
-              val clientEori: String = getClientEori(searchedAuthorities)
-
-              dataStoreService.getCompanyName(clientEori).map { companyName => {
-                Ok(authorisedToViewSearchResult(searchQuery, clientEori, searchedAuthorities, companyName, displayLink))}
-              }
-            }
+      }
+    }
+    result.flatten
+  }
+
+  /**
+   * Search and processes the Authorities for the valid input, further, displays the view accordingly
+   */
+  private def searchAuthoritiesForValidInput(request: AuthenticatedRequest[AnyContent],
+                                              searchQuery: EORI,
```

Comment:
same here

Reply:
indentation for the method name was fine but I reformatted the method parameters.Have a look

Reply:
Reviewer is happy with the change. Hence resolving.